etc/zabbix/zabbix_agentd.d/ic_azure.conf

```diff
@@ -1,10 +1,10 @@
-# Azure metric retrieval (configuration file, resource, metric, statistic, timegrain (optional: timeshift, instance name, role name)
-UserParameter=azure.metric[*],source /opt/virtualenv/zabbix-azure-monitoring/bin/activate && azure_metric "$1" "$2" "$3" "$4" "$5"
-UserParameter=azure.metric.timeshift[*],source /opt/virtualenv/zabbix-azure-monitoring/bin/activate && azure_metric "$1" "$2" "$3" "$4" "$5" --timeshift "$6"
-UserParameter=azure.metric.roleinstance[*],source /opt/virtualenv/zabbix-azure-monitoring/bin/activate && azure_metric "$1" "$2" "$3" "$4" "$5" --instance-name "$6"
-UserParameter=azure.metric.rolename[*],source /opt/virtualenv/zabbix-azure-monitoring/bin/activate && azure_metric "$1" "$2" "$3" "$4" "$5" --role-name "$6"
-UserParameter=azure.metric.roleinstance.timeshift[*],source /opt/virtualenv/zabbix-azure-monitoring/bin/activate && azure_metric "$1" "$2" "$3" "$4" "$5" --instance-name "$6" --timeshift "$7"
-UserParameter=azure.metric.rolename.timeshift[*],source /opt/virtualenv/zabbix-azure-monitoring/bin/activate && azure_metric "$1" "$2" "$3" "$4" "$5" --role-name "$6" --timeshift "$7"
+# Azure metric retrieval (configuration file, resource, metric, statistic, timegrain (optional: filter, metric-namespace, timeshift)
+UserParameter=azure.metric[*],source /opt/virtualenv/zabbix-azure-monitoring/bin/activate && azure_metric "$1" "$2" "$3" "$4" "$5" $([ -n "$6" ] && echo "--timeshift $6")
+UserParameter=azure.metric.filter[*],source /opt/virtualenv/zabbix-azure-monitoring/bin/activate && azure_metric "$1" "$2" "$3" "$4" "$5" --filter "$6" $([ -n "$7" ] && echo "--timeshift $7")
```

Comment:
I don't think general filter keys are actually usable since all possible quote marks are forbidden by Zabbix.

Reply:
I can't use single quote inside a double quote? Oh man, I thought I could. I need to test this.

Reply:
List of unsafe characters in user parameters: 
\ ' " ` * ? [ ] { } ~ $ ! & ; ( ) < > | # @

Reply:
These quoting issues were discussed in a chat and decided to fix using custom UserParameters. In commit bafe9d4110c808f68f828d1f21db315dcb14b89e.